homeassistant/components/rachio/switch.py

```diff
@@ -291,6 +291,7 @@ def _async_handle_update(self, *args, **kwargs) -> None:
             endtime = parse_datetime(args[0][0][KEY_RAIN_DELAY_END])
             _LOGGER.debug("Rain delay expires at %s", endtime)
             self._state = True
+            assert endtime is not None
```

Comment:
`parse_datetime` can return `None`, but `async_track_point_in_utc_time` doesn't allow it